### PR TITLE
Fix waveform misalignment for short audios

### LINKED ISSUE #1304: Very short audio tracks have misaligned progress bar upon track completion
## Description

Example: https://search-staging.openverse.engineering/audio/d4c5806f-1e86-4f41-835f-8fcda6c62cb7

The issue is very slight but it is there.

Originally reported by @zackkrida 

## Reproduction

1.  Visit https://search-staging.openverse.engineering/audio/d4c5806f-1e86-4f41-835f-8fcda6c62cb7 and play the track through to the end, note the slight misalignment of the progress bar at the end.

## Screenshots

image

## Additional context

We could potentially solve this by setting the progress bar width to exactly 100% of the waveform width once the track completes. This seems like it's probably a rounding error.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1304 by @sarayourfriend 

## Description

This PR fixes the waveform misalignment at the end of the waveform. It was especially noticeable for very short audios.   We were only updating the `currentTime` value when the audio was in `playing` status, so the last update when the audio went from `playing` to `played` status was never made, and the audio waveform `progressBarWidth` was not updated for the last time. 
There was also an error when we used a computed without calling its `.value`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
LGTM! I thought this would be an unsolvable problem related to web api rounding or some other issue. So glad there was a fix.

File: src/components/VAudioTrack/VAudioTrack.vue [hunk 1]
```diff
@@ -188,10 +188,13 @@ export default defineComponent({
...
-        currentTime.value = localAudio.currentTime
-        window.requestAnimationFrame(updateTimeLoop)
-      }
+      if (localAudio)
+        if (status.value === 'playing') {
+          currentTime.value = localAudio.currentTime
+          window.requestAnimationFrame(updateTimeLoop)
+        } else if (status.value === 'played') {
+          currentTime.value = localAudio.currentTime
+        }
```
Comment (sarayourfriend) [inline src/components/VAudioTrack/VAudioTrack.vue:197]: I'm going to look into why later but I wanted to note it here so I don't forget. It looks like the final update to put the audio current time happens slightly later and causes a jump in the waveform progression at the end for short audio files :thinking: I want to try to find a fix for that too if you don't mind @obulat?

**Review by sarayourfriend — APPROVED** [review]
Approving this for now. I haven't had time to check out the issue I mentioned. @obulat if you don't get around to checking on this now can you open an issue to address it? I think it's very low priority, but this fix is better to have than to block on it for the minor jumpiness.

**Comment by zackkrida** [comment]
@sarayourfriend does this video look like what you're seeing? I haven't been able to reproduce the jump you mentioned.

https://user-images.githubusercontent.com/6351754/172632555-af3422e0-c56a-4ef7-b8c9-5558f7becc8b.mp4

I have, however, @obulat, noticed interesting behavior with audio around 1 second long. Here's the 'end point' of a bunch of >1s audios:


CleanShot 2022-06-08 at 09 46 21@2x

They all finish in a different position! I wonder if the position is deterministic and related to their exact length. Perhaps for these we'd need to round up the progress bar to 1 second.

I also wonder: should we alter the timestamp to show milliseconds for audio tracks less than one second long? We can make a separate issue for that if needed.

**Comment by sarayourfriend** [comment]
@zackkrida it does not look like your video. Unfortunately I'm completely unable to play any audio files locally at the moment (no idea why, it's very strange) so I can't take a video to show what I was seeing.

**Comment by obulat** [comment]
> They all finish in a different position! I wonder if the position is deterministic and related to their exact length. Perhaps for these, we'd need to round up the progress bar to 1 second.

This is a really interesting bug! Sometimes I get the correct end position every time, sometimes it's incorrect on the first time, sometimes - on following replays. I'm going to look into it.

File: src/components/VAudioTrack/VAudioTrack.vue (hunk 1 above)
Comment (obulat) [reply src/components/VAudioTrack/VAudioTrack.vue]: I wonder if it was caused by the audio sometimes ending with a `paused` status, and sometimes with `played`. I still don't understand why it happens, but I removed the check for `status` being played to update the `currentTime` on line 196 here.

**Comment by obulat** [comment]
> I'm wondering if instead of relying on the animation loop we need to just set `currentTime = duration` when [the audio is finished playing (via the `ended` event)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/ended_event).

It seems to be working now, after removing the check for played status. Do you think this solution is good enough, @sarayourfriend ?

**Comment by zackkrida** [comment]
Looks better to me, nice 👍

**Comment by obulat** [comment]
> It makes me wonder if we really should only show milliseconds for sub-second files, mostly to minimize the visibility of that frustrating data issue 😬

This was an easy thing to fix :)

**Pull request merged**